node/src/neighborhood/overall_connection_status.rs

```diff
@@ -337,13 +337,14 @@ mod tests {
 
     #[test]
     #[should_panic(
-        expected = "Unable to receive node addr for the descriptor NodeDescriptor { blockchain: EthRopsten, encryption_public_key: 0x000000, node_addr_opt: None }"
+        expected = "Unable to receive node addr for the descriptor NodeDescriptor { blockchain: EthRopsten, encryption_public_key: 0x000000, node_addr_opt: None, country_code: \"ZZ\" }"
     )]
     fn can_not_create_a_new_connection_without_node_addr() {
         let descriptor_with_no_ip_address = NodeDescriptor {
             blockchain: Chain::EthRopsten,
             encryption_public_key: PublicKey::from(vec![0, 0, 0]),
             node_addr_opt: None,
+            country_code: "ZZ".to_string(),
```

Comment:
Why does a descriptor need to contain the country code? I cannot understand. I actually think it's even wrong. 
We need that descriptor just to contact another Node and this Node will respond with a NodeRecord about himself and there is the place where we should care about the country code for the first time. 

It should probably go away from inside this structure.